Added "Combining Data" chapter

grssnbchr opened the pull request:
This pull request adds a new chapter to the project, which illustrates various techniques for merging and combining datasets. The ideal position for this chapter would be after the "Reading Data" chapter.

grssnbchr committed:
added chapter for combining data, including first sub-chapter ("join data sets")

grssnbchr committed:
added last two subsections to combine data chapter

grssnbchr committed:
changed table of contents order

iros commented on locales/en/combine_data.md:
```diff
@@ -0,0 +1,75 @@
+
+## Add columns from different data sets
+
+In the last case, we have two or more data sets which were split *vertically*, in other words, attributes describing the same observations are stored in different data sets and need to be combined. This implies that all data sets have the same length. For example, `dataset_1` below contains two observations of attribute `a` and attribute `b`, while `dataset_2` contains the same two observations,  ...
```
Maybe make it a bit clearer that both objects reference to the same conceptual entity, so merging them makes sense because at every index of both datasets we'll have separate information about the same entity?

iros commented:
General comment: you might have noticed in other sections we add a little html snippet below the code snippet to be explicit about the libraries used in the code:

`<div class="aside">This code is using d3.js</div>`

Would you mind adding these appropriately? I know you mention it in the language of the post, but it would be more consistent. We can do both. Also, we only do this for non native functions.

Otherwise this looks great to me! Thanks for explaining this so well!

grssnbchr committed:
clarified last subsection, changed general wording, fixed a typo

grssnbchr commented:
@iros Concerning the asides, I've already added them for the last two code snippets. Apart from them, everything is native code. Thanks also for your input concerning the last section. I changed the example and the wording, so it should be more clear. I've also changed the wording in the section titles. 

By the way, I didn't commit any build output (in `public`), I hope this is okay.

grssnbchr committed:
seems like paths to data must be absolute..?

vlandham commented:
looks awesome. Thanks again @wnstnsmth for your contribution! 
If everyone is ok with it, I will attempt to merge and generate the `public` files.
I think we should probably remove `public` from the repo eventually - so its good that we don't have them generated.

vlandham commented:
Also, what would you think @iros and @wnstnsmth  if we shifted the position in the contents to be after 'summarizing data'. I understand that it doesn't depend on this - just that it might be a bit strange to jump immediately from one data set to 2 without first looking at them. Just a thought.
